controllers/machine_reconcile_scope.go

```diff
@@ -195,14 +207,17 @@ func (scope *machineReconcileScope) reconcile(hw *tinkv1.Hardware) error {
 }
 
 // patchHardwareStates patches a hardware's metadata and instance states.
-func (scope *machineReconcileScope) patchHardwareStates(hw *tinkv1.Hardware, mdState, iState string) error {
+func (scope *machineReconcileScope) patchHardwareStates(hw *tinkv1.Hardware) error {
 	patchHelper, err := patch.NewHelper(hw, scope.client)
 	if err != nil {
 		return fmt.Errorf("initializing patch helper for selected hardware: %w", err)
 	}
 
-	hw.Spec.Metadata.State = mdState
-	hw.Spec.Metadata.Instance.State = iState
+	for _, ifc := range hw.Spec.Interfaces {
+		if ifc.Netboot != nil {
+			ifc.Netboot.AllowPXE = ptr.To(false)
+		}
+	}
```

Comment:
I think the caller should be setting this. 

`patchHardwareStates` has no business logic that describes the conditions for assigning false so its peculiar to have it live here. We also previously made the decision in the calling function and I think that's the right level to have the logic.